extensions/markdown/src/extension.ts

```diff
@@ -0,0 +1,174 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+ 'use strict';
+
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+import * as path from 'path';
+import { ExtensionContext, TextDocumentContentProvider, EventEmitter, Event, Uri, TextDocument, ViewColumn } from "vscode";
+
+const hljs = require('highlight.js');
+const mdnh = require('markdown-it-named-headers');
+const md = require('markdown-it')({
+    html: true,
+    highlight: function (str, lang) {
+        if (lang && hljs.getLanguage(lang)) {
+            try {
+                return `<pre class="hljs"><code><div>${hljs.highlight(lang, str, true).value}</div></code></pre>`;
+            } catch (error) { }
+        }
+        return `<pre class="hljs"><code><div>${md.utils.escapeHtml(str)}</div></code></pre>`;
+    }
+}).use(mdnh, {});
```

Comment:
If you don't mind me asking.  As it is coded here, anytime a member wants to add a plugin to markdown-it, it seems they will have to:

1. `Npm install markdown-it-plugin --savedev` in extensions folder
2.  Require package here and add package to `.use(...)`

This seems fine to me, but I believe that also means I would have to do this _every_ time that extension is updated.  I don't know of a good solution, but one solution I thought of is to allow for developers to create a "skeleton" package that simply exposes and api to the markdown extension that says use this.  So people will be able to install a package, say: `markdown-it-katex` for math support which has a dependency on this package and it should just work.